cpp/src/converters/renumber.cuh

```diff
@@ -171,20 +105,18 @@ void renumber_vertices(size_t size,
 
   renumber::hash_type hash_size = hash.getHashSize();
 
-  T_in *hash_data;
-
-  renumber::index_type *hash_bins_start;
-  renumber::index_type *hash_bins_end;
+  rmm::device_vector<T_in> hash_data_v(2 * size);
+  rmm::device_vector<renumber::index_type> hash_bins_start_v(1 + hash_size,
+                                                             renumber::index_type{0});
+  rmm::device_vector<renumber::index_type> hash_bins_end_v(1 + hash_size);
 
-  ALLOC_TRY(&hash_data, 2 * size * sizeof(T_in), stream);
-  ALLOC_TRY(&hash_bins_start, (1 + hash_size) * sizeof(renumber::index_type), stream);
-  ALLOC_TRY(&hash_bins_end, (1 + hash_size) * sizeof(renumber::index_type), stream);
+  T_in *hash_data                       = hash_data_v.data().get();
+  renumber::index_type *hash_bins_start = hash_bins_start_v.data().get();
+  renumber::index_type *hash_bins_end   = hash_bins_end_v.data().get();
```

Comment:
Just FYI, thrust has raw pointer cast I started to use this (i.e. thrust::raw_pointer_cast(hash_data_v.data()), I think this makes the intention (casting a thrust::device_ptr to a raw pointer) clearer.

Reply:
I used that construct when I started the cudf++ work and Jake preferred this syntax... so I've been using this syntax instead.  It's certainly a construct we use a great deal, we should probably come to some agreement on what we prefer.

Reply:
> I used that construct when I started the cudf++ work and Jake preferred this syntax... so I've been using this syntax instead. It's certainly a construct we use a great deal, we should probably come to some agreement on what we prefer.

Good to know! I just searched cudf codebase for raw_pointer_cast and data().get() and raw_pointer_cast appeared only few times except for the legacy parts and data().get() is dominantly used.

Better just follow cudf's convention, and I will switch back to data().get() :-)